packages/app/src/systems/Account/services/account.ts

```diff
@@ -107,22 +108,45 @@ export class AccountService {
     try {
       const provider = await createProvider(providerUrl!);
       const balances = await getBalances(provider, account.address);
-      const balanceAssets = await AssetsCache.fetchAllAssets(
-        provider.getChainId(),
+      const assetsAmountsInUsd: Record<string, string | undefined> = {};
+      const chainId = provider.getChainId();
+
+      const balanceAssets = AssetsCache.fetchAllAssets(
+        chainId,
         balances.map((balance) => balance.assetId)
       );
+      let totalBalanceInUsd = 0;
+
+      const assetsAmountsUsdPromise = balances.map((asset) => {
+        return convertAsset(
+          chainId,
+          asset.assetId,
+          asset.amount.toString()
+        ).then((rate) => {
+          assetsAmountsInUsd[asset.assetId] = rate?.amount;
+          if (rate?.amount) {
+            totalBalanceInUsd += Number.parseFloat(
+              rate.amount.replace(/[$,]/g, '')
+            );
+          }
+        });
+      });
+      await Promise.all([...assetsAmountsUsdPromise, balanceAssets]);
```

Comment:
1- we can use the rate we already have in the assets
2- if we would use this code it would be better to do the .then part later after the Promise.all completed.